libnavigation-core/src/main/java/com/mapbox/navigation/core/NavigationSession.kt

```diff
@@ -18,7 +21,7 @@ internal class NavigationSession(private val context: Context) : RoutesObserver,
             val previousValue = state
             field = value
 
-            // todo expose state observers for the rest of the lib to hook into?
+            stateObservers.forEach { it.onNavigationSessionStateChanged(value) }
 
             when {
                 previousValue == State.ACTIVE_GUIDANCE -> MapboxNavigationAccounts.getInstance(
```

Comment:
Since we do have an observer now, we should extract the accounts lib logic to a separate class.